wheatley/tower.py

```diff
@@ -216,69 +212,66 @@ def _on_user_leave(self, data):
         # Unassign all instances of that user
         for bell, user in self._assigned_users.items():
             if user == user_id_that_left:
-                self._assigned_users[bell] = None
+                del self._assigned_users[bell]
```

Comment:
This doesn't work, it's changing the dictinary that's being enumerated.

I prefer the old way here and in `_on_assign_user` of
```python
self._assigned_users[bell] = None
```
Which needs a change of signature to 
```python
self._assigned_users: Dict[Bell, Optional[int]] = {}
```

Reply:
I always thought that having two ways to represent unassigned bells was a bit janky - but I think it's more aggrievous to have to work round the changing-dict-while-enumerating error.

Reply:
I think RingingRoom sends the assignments for all the bells so we would start with a complete dict? Ah but then if the size increases we don't add new ones.

Yes, that sounds ok. Possibly a race condition, but in practice I can't quite work out how you could get two assignments on the same bell happening close enough together. Really I think it's between a one liner in both sites vs avoiding having some unassigned bells not in the dictionary.

Reply:
Cool! Fixed now.

Also if race conditions end up an issue, then we could assign bells to `None` temporarily (overriding the type checker), and then filter out the `None`s later, but that is unnecessarily complicated and so I wouldn't do it unless forced.